scripts/quests/crystalWar/The_Lost_Book.lua

```diff
@@ -0,0 +1,198 @@
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED and vars.Prog == 2
+        end,
+
+        [xi.zone.WINDURST_WATERS_S] =
+        {
+            ['Door_Rhinostery_South'] = quest:progressEvent(144),
+
+            onEventFinish =
+            {
+                [144] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 3)
+                end,
+            },
+        },
+    },
+
+    -- Go to the Optistery (F-8 on the first map) of Windurst (S) and examine the door for a cutscene.
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED and vars.Prog == 3
+        end,
+
+        [xi.zone.WINDURST_WATERS_S] =
+        {
+            ['Door_Optistery']        = quest:progressEvent(145),
+            ['Door_Rhinostery_South'] = quest:event(150), -- Optional dialogue
+
+            onEventFinish =
+            {
+                [145] = function(player, csid, option, npc)
+                    player:delKeyItem(xi.ki.LEATHERBOUND_BOOK)
+                    quest:setVar(player, 'Prog', 4)
+                end,
+            },
+        },
+    },
+
+    -- Head to Castle Oztroja (S) and examine the ??? at (G-8) on the first map.
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED and vars.Prog == 4 and
+                not player:hasKeyItem(xi.ki.LYNX_PELT)
+        end,
+
+        [xi.zone.CASTLE_OZTROJA_S] =
+        {
+            ['_qm0'] =
+            {
+                onTrigger = function(player, npc)
+                    npcUtil.giveKeyItem(player, xi.ki.LYNX_PELT)
+                end,
+            },
+        },
+    },
+
+    -- Trade a sheet of Vellum to the Optistery door in Windurst Waters (S).
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED and vars.Prog == 4 and
+                player:hasKeyItem(xi.ki.LYNX_PELT)
+        end,
+
+        [xi.zone.WINDURST_WATERS_S] =
+        {
+            ['Door_Optistery'] =
+            {
+                onTrade = function(player, npc, trade)
+                    if npcUtil.tradeHasExactly(trade, xi.items.VELLUM) then
+                        return quest:progressEvent(146)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [146] = function(player, csid, option, npc)
+                    player:confirmTrade()
+                    player:delKeyItem(xi.ki.LYNX_PELT)
+                    quest:setVar(player, 'Prog', 5)
+                    quest:setVar(player, "Timer", VanadielUniqueDay() + 1)
+                end,
+            },
+        },
+    },
+
+    -- Waited game day
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED and vars.Prog == 5 and
+                quest:getVar(player, "Timer") <= vanaDay()
```

Comment:
Should probably be consistent here as well, since we set it to VanadielUniqueDay(), compare it to it as well.